enzyme/Enzyme/MLIR/Passes/InlineEnzymeRegions.cpp

```diff
@@ -180,28 +192,40 @@ outlineAutoDiffFunc(enzyme::AutoDiffRegionOp op, StringRef funcName,
 }
 
 LogicalResult outlineEnzymeAutoDiffRegion(enzyme::AutoDiffRegionOp op,
-                                          StringRef defaultFuncName,
+                                          StringRef funcName,
                                           OpBuilder &builder) {
-  StringRef funcName = op.getFn().value_or(defaultFuncName);
   OpBuilder::InsertionGuard insertionGuard(builder);
   builder.setInsertionPointAfter(op->getParentOfType<SymbolOpInterface>());
 
   SmallVector<enzyme::Activity> argActivities =
       llvm::map_to_vector(op.getActivity().getAsRange<enzyme::ActivityAttr>(),
                           [](auto attr) { return attr.getValue(); });
-
-  size_t numPrimalArgs = op.getActivity().size();
+  size_t numSeeds = llvm::count_if(
+      op.getRetActivity().getAsRange<enzyme::ActivityAttr>(), [](auto attr) {
+        switch (attr.getValue()) {
+        case enzyme::Activity::enzyme_active:
+        case enzyme::Activity::enzyme_activenoneed:
+          return true;
+        default:
+          return false;
+        }
+      });
+  size_t numPrimalArgs = op.getInputs().size() - numSeeds;
```

Comment:
Hm I don't fully understand this. Why is get activity size not equal to the number of primal args,? That should always be the case, right?

Reply:
I agree the name is confusing. `numPrimalArgs` and `primalArgs` actually refers to the number of primal arguments and any duplicated shadow args. So for the added test, we have `enzyme.autodiff_region(%x, %dx, %seed)`. Crucially, we want to insert any free variables *after* `%dx` and *before* `%seed`.

I considered renaming to `nonSeeds`, but I'm not sure.